src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i !== this.length - 1) {
+        if (this[i] === null || this[i] === undefined) {
+          result += separator;
+        } else {
+          result += this[i];
+          result += separator;
+        }
+      } else {
+        result += this[i];
```

Comment:
You need to check all the elements for null and undefined. Now you don't check the last one